src/internal/snapshot/api_server.go

```diff
@@ -0,0 +1,72 @@
+// Package snapshot implements subsystem-independent disaster recovery.
+package snapshot
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/pachyderm/pachyderm/v2/src/internal/dbutil"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/log"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+	"github.com/pachyderm/pachyderm/v2/src/internal/snapshotdb"
+	"github.com/pachyderm/pachyderm/v2/src/snapshot"
+)
+
+type apiServer struct {
+	snapshot.UnimplementedAPIServer
+	env Env
```

Comment:
I wouldn't bother with the indirection through env, or making this struct private.  Callers can easily:

```go
server := &snapshot.APIServer{
    DB: db, 
    Store: store,
}
```
We usually use Env to break dependency cycles, but there are none here.

Reply:
When hooking the server into pachd, I find these env are still necessary? Other severs are passing around the envs